xcm/xcm-simulator/example/src/lib.rs

```diff
@@ -212,4 +212,39 @@ mod tests {
 			);
 		});
 	}
+
+	#[test]
+	fn recursion_limit_works() {
+		use parachain::{Event, System};
+		MockNet::reset();
+
+		ParaA::execute_with(|| {
+			let msg = WithdrawAsset { assets: vec![All], effects: vec![] };
```

Comment:
let's add a test for a recursion of `WithdrawAssets { BuyExecution { WithdrawAsset { ... }}}`?